controller/iteration.go

```diff
@@ -172,23 +184,42 @@ func (c *IterationController) Update(ctx *app.UpdateIterationContext) error {
 		return jsonapi.JSONErrorResponse(ctx, goa.ErrNotFound(err.Error()))
 	}
 
-	return application.Transactional(c.db, func(appl application.Application) error {
-		itr, err := appl.Iterations().Load(ctx.Context, id)
+	var itr *iteration.Iteration
+	var sp *space.Space
+	var editorIsCreator bool
+	err = application.Transactional(c.db, func(appl application.Application) error {
```

Comment:
I wonder if these two functions can be somehow combined and outsourced into one:

from above ([here](https://github.com/fabric8-services/fabric8-wit/pull/1783/files#diff-26150d9e0df4f4f8a72b25cdaee00342R61))
```go
  		parent, err = appl.Iterations().Load(ctx, parentID)
  		if err != nil {
			return err
  		}		  		
		sp, err = appl.Spaces().Load(ctx, parent.SpaceID)
  		if err != nil {
			return jsonapi.JSONErrorResponse(ctx, err)
  		}		  		
		if uuid.Equal(*currentUser, sp.OwnerID) {
			creatorIsSpaceOwner = true
			return nil	
  		}
```

[here](https://github.com/fabric8-services/fabric8-wit/pull/1783/files#diff-26150d9e0df4f4f8a72b25cdaee00342R191)
```go
		itr, err = appl.Iterations().Load(ctx.Context, id)
  		if err != nil {
			return err
  		}
		sp, err = appl.Spaces().Load(ctx, itr.SpaceID)
  		if err != nil {
			return err
  		}
  		if uuid.Equal(*currentUser, sp.OwnerID) {
 			editorIsCreator = true
 			return nil
 		}
 		return nil
```

Reply:
I would even suggest that this piece of code moves to the `authz.Authorize`. This is a bigger refactoring, but it could delegate the whole logic of permission in a single place instead of spreading it between this controller (when checking if the user owns the space) and the `authz.Authorize` func (when the user is a collaborator in this space). WDYT ?

Reply:
@xcoulon @kwk a thought on that - I can update `auth.Authorize()` function to return three things instead of two
`(isOwner bool, isCollaborator bool, err error)`
But iteration loading will be from controller only, as `Authorize` has nothing to do with it as of now, when we have `ACL` or `better Role based auth` then the code may look different?
Will it be okay to go ahead with this?

Reply:
The entire authorization flow will be refactored. Not sure when though :) Will take months.

Reply:
@pranavgore09 not sure I understand your concern: the idea was to check if the user is the owner or collaborator of a given space, just as you wrote above. Then this function could be reused anywhere else, when we need to check the permissions.

Reply:
@xcoulon please see https://github.com/fabric8-services/fabric8-wit/pull/1783/commits/052c95d5c06dd823250832484307d8204635d2ec